libs/designer/src/lib/core/parsers/BJSWorkflow/BJSDeserializer.ts

```diff
@@ -360,7 +360,7 @@ const addActionsInstanceMetaData = (nodesMetadata: NodesMetadata, runInstance: L
   const updatedNodesData = { ...nodesMetadata };
 
   Object.entries(updatedNodesData).forEach(([key, node]) => {
-    const nodeRunData = runInstanceActions[key];
+    const nodeRunData = runInstanceActions && runInstanceActions[key];
```

Comment:
Fun fact (or nit) optional chaining works with [ ] property references too, so you could do this and it makes it null safe on both sides, returning undefined if either fails
```suggestion
    const nodeRunData = runInstanceActions?.[key];
```

Reply:
Thanks, I will update it